QEfficient/transformers/models/modeling_auto.py

```diff
@@ -366,11 +367,15 @@ def generate(self, prompts: List[str], device_id: List[int] = None, runtime: str
         ``optional`` Args:
             :runtime (str, optional): Only ``AI_100`` runtime is supported as of now; ``ONNXRT`` and ``PyTorch`` coming soon. Defaults to "AI_100".
         """
-        assert Runtime(runtime) == Runtime.AI_100, "Only AI_100 runtime is supported right now via generate API"
+        if Runtime(runtime) != Runtime.AI_100:
+            raise ValueError("Only AI_100 runtime is supported right now via generate API")
         self.run_cloud_ai_100(prompts=prompts, device_id=device_id, **kwargs)
 
-    def run_cloud_ai_100(self, prompts: List[str], device_id: List[int] = None, **kwargs):
-        assert isinstance(self.qpc_path, str), "Please run compile API first!"
+    def run_cloud_ai_100(self, prompts: List[str],  device_id: List[int] = None, **kwargs):
+        if not isinstance(self.qpc_path, str):
+            raise TypeError("Please run compile API first!")
+        if isinstance(self.device_id, type(None)):
+            raise TypeError("please pass valid device_id as input argument")
```

Comment:
I don't think we need check for device id, as if its none, I believe it's taken from auto device picker. Please check this once.

Reply:
Yes, no need to check `device_id` for none.

Reply:
Removed, thanks.